### ISSUE TITLE
Optional elements can not be parsed further

### ISSUE THREAD

**Issue by ruudvdd** [alexdebril/feed-io#76]
This issue is similar to issue #72 , but more generic.

RSS 2.0 feeds can be extended with other elements than specified in the specification. See http://www.rssboard.org/rss-specification#extendingRss .

When this namespace contains tags with subtags, feed-io is not capable of reading those subtags separately.

For example:

XML:
```
<itunes:owner>
    <itunes:name>Example</itunes:name>
    <itunes:email>info@example.org</itunes:email>
</itunes:owner>
```

PHP:
```
$ownerValue = $feed->getValue('itunes:owner');
var_dump($ownerValue);
```

Output:
```
string(58) "
            Example
            info@example.org
        "
```

I'm pretty sure (but I can be wrong) feed-io is not capable of fetching the content of itunes:name.

I have two suggestions to handle this problem:

1) Allow developers to create their own Feed Rulesets and Objects to add to the feed-io parser as an extension

2) Modify the FeedIo\Feed\Node\Element to store and parse subtags of that element

What's your vision on this problem? Is this something feed-io should be able to do in the future?

**Comment by alexdebril** [comment]
Hi,

Thank you for your feedback, your point of view sounds interesting to me.

When I designed feed-io, I preferred to focus on the main properties of a feed : description, title, date and so on. Even the author was somehow left behind because I've noticed that many feeds don't provide it. Now that users are asking for more properties, I have to find a solution for it. My answers to your suggestions : 

1 It's already possible to add custom ruleSets in feed-io. I just didn't document it and I'll do it shortly to let developers extend it with their own rules

2 yes, FeedIo\Feed\Node\Element needs that. And I think it will lead to a "generic" format able to parse and read everything.

Stay tuned, it may come quickly

**Comment by ruudvdd** [comment]
Hi,

I already worked on it on a fork. I have not pushed it yet to my fork, but the parsing is already done. I'm now working on the writing of those sub elements. I'll try to send a pull request this week.

**Comment by alexdebril** [comment]
ok, thank you very much.

Are you adding the ability for FeedIo\Feed\Node\Element to hold other instances of itself ?

Something like : 

```php

class Element implements ElementInterface
{

protected $subElements;

public function addElement(Element $element) {
     $this->subElements[] = $element;
}

public function getElements() {
    return $this->subElements;
}
```

It would be great because it could handle an unlimited numbers of sub elements.

Thanks !

**Comment by ruudvdd** [comment]
Yes, that was exactly the thing I'm doing.

**Comment by alexdebril** [comment]
:+1:

**Comment by ruudvdd** [comment]
Thanks for merging. Is it possible to include it in the next release?

**Comment by alexdebril** [comment]
Yes, I just wanted to make some tests before the next release. It may come
quickly.

**Comment by ruudvdd** [comment]
Ok, great. Tell me if I can help

**Comment by alexdebril** [comment]
Thank you, it's included in the 2.5 release
